pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -374,5 +375,5 @@ func (r *Reconciler) Reconcile(request reconcile.Request) (reconcile.Result, err
 	}
 
 	logger.Info("Starting the bind of application(s) with backing service...")
-	return r.bind(logger, binder, secret, retrievedData, sbr, sbrStatus, objectsToAnnotate)
+	return r.bind(logger, binder, secret, retrievedData, retrievedCache, sbr, sbrStatus, objectsToAnnotate)
```

Comment:
Same idea here, make sure `retreievedData` contains it all.

Reply:
> Same idea here, make sure `retreievedData` contains it all.

But retrievedData maintains a key,value pair list whereas for cache we need to retain the whole structure (hierarchy) as mentioned here https://github.com/redhat-developer/service-binding-operator/issues/272#issuecomment-562735404

Reply:
Having just the `secretData` would not maintain that structure